src/main/java/com/example/bookstore/service/impls/OrderServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+package com.example.bookstore.service.impls;
+
+import com.example.bookstore.dto.OrderPlacedConfirmation;
+import com.example.bookstore.dto.request.OrderRequestDto;
+import com.example.bookstore.dto.request.OrderStatusRequestDto;
+import com.example.bookstore.dto.response.OrderResponseDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.OrderMapper;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.Order;
+import com.example.bookstore.model.OrderItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.CartItemRepository;
+import com.example.bookstore.repository.OrderItemRepository;
+import com.example.bookstore.repository.OrderRepository;
+import com.example.bookstore.repository.ShoppingCartRepository;
+import com.example.bookstore.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderMapper orderMapper;
+
+    @Override
+    public OrderPlacedConfirmation create(User user, OrderRequestDto dto) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUser_Id(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "You're shopping cart is empty, "
+                                + "you have to place good's into the shopping cart"));
+        Order orderFromDb = setUpAndSaveOrder(user, dto, shoppingCart);
+        deleteCartItemsCreateOrderItems(shoppingCart, orderFromDb);
```

Comment:
setUp**And**SaveOrder
deleteCartItems**And**CreateOrderItems

this is not a Single responsibility